components/raftstore/src/store/fsm/peer.rs

```diff
@@ -5247,6 +5273,53 @@ where
                         self.register_pull_voter_replicated_index_tick();
                     }
                 }
+                ExecResult::ForceCompact { apply_state } => {
+                    let last_index = apply_state.get_truncated_state().index;
+                    let first_index = self.fsm.peer.raft_group.raft.r.raft_log.first_index();
+
+                    {
+                        let peer_store = self.fsm.peer.mut_store();
+                        // self.on_ready_compact_log(first_index,
+                        // apply_state.get_truncated_state().clone());
+                        peer_store.set_apply_state(apply_state);
+                        peer_store.clear_entry_cache_warmup_state();
+                        peer_store.compact_entry_cache(last_index + 1);
+                        peer_store.raft_state_mut().mut_hard_state().commit = last_index;
+                        peer_store.raft_state_mut().last_index = last_index;
+                    }
+
+                    let raft_engine = self.fsm.peer.get_store().raft_engine();
+                    let mut batch = raft_engine.log_batch(2);
+                    raft_engine
+                        .gc(self.region_id(), first_index, last_index, &mut batch)
```

Comment:
shouldn't introduce other io operation in raftstore thread

Reply:
After discussing offline, we decide to keep the current code to make the logic simpler. We maybe enhance it in the future if this operation is still a thing.